src/modules/src/Eryph.Modules.Controller/ChangeTracking/ChangeTrackingSaveChangesInterceptor.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using LanguageExt;
+using Microsoft.EntityFrameworkCore.Diagnostics;
+using Microsoft.Extensions.Logging;
+
+namespace Eryph.Modules.Controller.ChangeTracking;
+
+internal class ChangeTrackingSaveChangesInterceptor<TChange>(
+    IChangeDetector<TChange> detector,
+    IChangeTrackingQueue<TChange> queue,
+    ILogger logger)
+    : SaveChangesInterceptor
+{
+    private Seq<ChangeTrackingQueueItem<TChange>> _currentItem = Prelude.Empty;
+
+    public override async ValueTask<InterceptionResult<int>> SavingChangesAsync(
+        DbContextEventData eventData,
+        InterceptionResult<int> result,
+        CancellationToken cancellationToken = default)
+    {
+        // EF Core 7 and later does not create database transactions when saving
```

Comment:
EF Core 7 and newer avoids to create transactions when saving changes. Afaik,
this happens when the changes can be performed with a single database statement.
Hence, we need two separate interceptions now: when saving changes and when
committing transactions. According to the EF Core docs, this has a performance
advantage.

Alternative, we can configure EF Core to always create a transaction when saving
changes (`AutoTransactionBehavior.Always`). This way, we do not need the interception
when saving changes. @fw2568, we need to decide in general if we want/need transactions
on the database. If we think so, it might be better to use `AutoTransactionBehavior.Always`
as most operations will anyway happen in a transaction and the code is less complex.

Reply:
I don't think that we have to max performance here - and - when I consider rebus - I'm sure that we have to use a transaction in any case. So I would also would use AutoTransactionBehavior.Always.